### PR TITLE
code review #1

### PR DESCRIPTION
**Pull request by HypeDis**

### CODE REVIEW

File: server/db/models/review.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  content: {
+    type: TEXT,
+    allowNull: false,
+    len: {
+      args: [20] //this should be min characters
+    }
+  },
+  rating: {
+    type: INTEGER,
+    validate: {
```
Comment (peetklecha) [inline server/db/models/review.js:15]: maybe allowNull and just have min: 1?
and default null?
nbd

File: server/db/models/product.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const uuidDef = {
+  type: UUID,
+  primaryKey: true,
+  defaultValue: UUIDV4
+}
+
+const Product = db.define('product', {
+  id: uuidDef,
+  alcohol: {
+    type: ENUM ('Beer', 'Wine')
```
Comment (peetklecha) [inline server/db/models/product.js:14]: categories vs supercategories
make Category table recursive?
each category has a parentCategoryId
Category.hasMany(Category)?

File: server/db/models/product.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  description: {
+    type: TEXT,
+    allowNull: false,
+    validate: {
+      notEmpty: true
+    }
+  },
+  price: {
+    type: DECIMAL(10, 2),
```
Comment (peetklecha) [inline server/db/models/product.js:31]: alternative: could use INTEGER and store prices as pennies. up to you!

File: server/api/products.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    .catch(next)
+})
+
+router.get('/:id', (req, res, next) => {
+  Product.findByPk(req.params.id)
+    .then(thisProduct => res.send(thisProduct))
+    .catch(next)
+})
+
+//tyring this out
```
Comment (peetklecha) [inline server/api/products.js:18]: redundant?

File: server/api/products.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  Product.findAll({
+    where: {
+      categoryId: req.params.id
+    }
+  })
+    .then(products => res.send(products))
+    .catch(next)
+})
+
+router.post('/', (req, res, next) => {
```
Comment (peetklecha) [inline server/api/products.js:29]: protect your stuff!!!

File: server/db/models/review.js (hunk 1 above)
Comment (rcntype) [reply server/db/models/review.js:15]: Changed min to 1 and added allowNull to true which should make default null.  In the latest merge.

File: server/api/products.js (hunk 4 above)
Comment (rcntype) [reply server/api/products.js:18]: Deleted

File: server/api/orders.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+//get all orders
+router.get('/', (req, res, next) => {
+  Order.findAll()
+    .then(orders => res.send(orders))
+    .catch(next)
+})
+
+//get one specific order
+router.get('/:id', (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (rcntype) [reply server/api/orders.js]: Gone

File: client/components/Categories.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+const mapStateToProps = ({products}, ownProps) => {
+  const filteredProducts = products.filter(
+    product => product.categoryId === ownProps.match.params.id
+  )
+  return {
+    filteredProducts
+  }
+}
+
+export default connect(mapStateToProps, null)(Categories)
```
[the remark this answers is not in the document]
Comment (rcntype) [reply client/components/Categories.js:28]: Got it.

**Pull request merged**